src/services/chain.ts

```diff
@@ -325,8 +325,11 @@ export class ChainContext {
     log.debug(`Watchdog timeout: ${watchdogTimeout} seconds`);
     let lastBlockReceived = lastProcessedBlock;
     provider.on("block", async (blockNumber: number) => {
+      // Intentionally don't guard the getBlock call with a try/catch block
+      // because we want to crash the process if this fails. This will result
+      // in the kubernetes pod restarting and trying to recover.
```

Comment:
i would rephrase and not reference kubernetes here, as this is code that can be run in various ways